addons/web_editor/static/lib/odoo-editor/src/commands/enter.js

```diff
@@ -46,43 +45,46 @@ HTMLElement.prototype.oEnter = function (offset, firstSplit = true) {
     while (offset < this.childNodes.length) {
         splitEl.appendChild(this.childNodes[offset]);
     }
-    if (isBlock(this) || splitEl.hasChildNodes()) {
-        this.after(splitEl);
-        if (isVisible(splitEl)) {
-            didSplit = true;
-        } else {
-            splitEl.remove();
-        }
-    }
+
+    // Move spliEl after current node.
+    (isBlock(this) || !(this.nextSibling && this.nextSibling.nodeName === '#TEXT')) && this.after(splitEl);
 
     // Propagate the split until reaching a block element (or continue to the
     // closest list item element if there is one).
     if (!isBlock(this) || (this.nodeName !== 'LI' && this.closest('LI'))) {
         if (this.parentElement) {
-            this.parentElement.oEnter(childNodeIndex(this) + 1, !didSplit);
+            this.parentElement.oEnter(childNodeIndex(this) + 1, false);
         } else {
             // There was no block parent element in the original chain, consider
             // this unsplittable, like an unbreakable.
             throw UNBREAKABLE_ROLLBACK_CODE;
         }
     }
-
     // All split have been done, place the cursor at the right position, and
     // fill/remove empty nodes.
-    if (firstSplit && didSplit) {
+    if (firstSplit) {
         restore();
 
         fillEmpty(clearEmpty(this));
         fillEmpty(splitEl);
-
-        const focusToElement =
-            splitEl.nodeType === Node.ELEMENT_NODE && splitEl.tagName === 'A'
-                ? clearEmpty(splitEl)
-                : splitEl;
-        setCursorStart(focusToElement);
+        setCursorStart(splitEl);
     }
     return splitEl;
 };
+/**
+ * Specific behavior for anchor tag: do not split in two if cursor at the end but
+ * instead create a paragraph.
+ * Cursor end of line: <p><a>title[]</a></p> + ENTER <=> <p><a>title[]</a><p>[]<br/></p>
+ * Cursor in the line: <p><a>tit[]le</a></p> + ENTER <=> <p><a>tit</a></p><p><a>[]le</a></p>
+ */
+HTMLAnchorElement.prototype.oEnter = function () {
+    //do not split in two if cursor at the end but instead create a paragraph.
+    const newEl = HTMLElement.prototype.oEnter.call(this, ...arguments);
+    if (!newEl.textContent.replace('\u200B', '')) {
+        newEl.after(newEl.firstChild);
+        newEl.remove();
+    }
+}
```

Comment:
@dhba-odoo i dont think we need to create extra prototype for an inline tag, simply handle it in HTMLElement.